pkg/restarter/restarter.go

```diff
@@ -191,7 +191,7 @@ func (c *Controller) processEndpoint(key string) error {
 		return nil
 	}
 
-	ep, err := c.clientset.CoreV1().Endpoints(namespace).Get(name, metav1.GetOptions{})
+	ep, err := c.clientset.CoreV1().Endpoints(namespace).Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
Making a context in `processWorkItem` and passing to `processEndpoint` helps reuse the context.

Reply:
After the escape from vendor hell, my brain was too frayed to do this 😅 
:+1: Will make the change.